### PR TITLE
fix: Enable copyFrom const track proxy for MultiTrajectory

### PR DESCRIPTION
**Pull request by benjaminhuth**
This adds a `friend class` statement in the `MultiTrajectory` that enables `copyFrom(...)` from a `ConstTrackStateProxy` to a mutable `TrackStateProxy`.

I added this to the UnitTest, but this for now only checks if this compiles.

### CODE REVIEW

File: Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp [hunk 1]
```diff
@@ -868,4 +868,19 @@ BOOST_AUTO_TEST_CASE(ProxyAssignment) {
...
+BOOST_AUTO_TEST_CASE(CopyFromConst) {
+  using PM = TrackStatePropMask;
+  MultiTrajectory mj;
+
+  const auto idx_a = mj.addTrackState(PM::All);
+  const auto idx_b = mj.addTrackState(PM::All);
+
+  auto mutableProxy = mj.getTrackState(idx_a);
+  auto constProxy =
+      static_cast<const MultiTrajectory&>(mj).getTrackState(idx_b);
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp:881]: Does a static cast do the right thing here? I'd probably created an auxiliary variable `const MultiTrajectory&` to test this.

File: Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp [hunk 2]
```diff
@@ -868,4 +868,19 @@ BOOST_AUTO_TEST_CASE(ProxyAssignment) {
   // MultiTrajectory::TrackStateProxy tp4{tp3};
```
Comment (paulgessinger) [inline Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp:868]: Did you check that this line still doesn't compile? (the compilation failure is what is expected).

Also did you test that `copyFrom` does **NOT** work when called on a const track state proxy?

Comment (benjaminhuth) [reply Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp:868]: Yes, both lines still do not compile.

I also quickly tried to find a way to somehow turn such a compile error in a checkable `constexpr bool` but that seems not to be so easy unfortunately...

File: Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp (hunk 1 above)
Comment (benjaminhuth) [reply Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp]: It should be equivalent, but maybe its better to be more explicit here

File: Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp (hunk 2 above)
Comment (paulgessinger) [reply Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp:868]: Ok perfect!

**Review by paulgessinger — APPROVED** [review]
Ok, LGTM!

**Pull request merged**